connect/runtime/src/main/java/org/apache/kafka/connect/runtime/isolation/PluginUtils.java

```diff
@@ -16,26 +16,39 @@
  */
 package org.apache.kafka.connect.runtime.isolation;
 
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
 import java.io.IOException;
 import java.lang.reflect.Modifier;
-import java.net.URL;
 import java.nio.file.DirectoryStream;
 import java.nio.file.Files;
 import java.nio.file.Path;
 import java.util.ArrayList;
+import java.util.Arrays;
 import java.util.Collection;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.LinkedList;
 import java.util.List;
 import java.util.Locale;
+import java.util.Set;
 
 public class PluginUtils {
+    private static final Logger log = LoggerFactory.getLogger(PluginUtils.class);
+
     private static final String BLACKLIST = "^(?:"
             + "java"
             + "|javax"
+            + "|sun"
```

Comment:
I think the risk with things like this in the blacklist is if we can't guarantee that everything under the package will be provided by the system classpath loader, then doing this is kind of risky. In other words, it seems like we *must* be conservative in what we blacklist here. We have complete control over `o.a.k.*` so those ones definitely seem fine. `slf4j` definitely *seems* like a good idea, although tbh I would have to scan through what else could be included there. `java` I'm pretty sure is guaranteed to be provided by the system. Others like `sun.` and `org.omg` I'm not so sure about. What are the consequences of removing some of these? Do some connectors break if we limit this blacklist to just a few key packages that we're confident about?